### PR TITLE
Minimum purchase of 6 months, improved starknetid emptiness checks and whitelist removal

### PR DESCRIPTION
**Pull request by Th0rgal**
This first update to the naming contract adds:

- A minimum requirement of 6 months for purchasing and renewing a domain.
Here is an example on testnet with a modified client which allows 0.1 years:
image

- A better check to know if you can write on a StarknetID: it allows you to rewrite a domain on a StarknetID if its domain expired. @0xBenaparte should we even keep a check for this? If we don't someone would be able to lose his domain ownership until it expires by calling the contract manually (given we already check on the client). This would mean less code, so a safer contract and it would be cheaper. 

It removes:
- Whitelisting related code.

### CODE REVIEW

**Comment by Th0rgal** [comment]
fix: #13

**Comment by 0xBenaparte** [comment]
So if I write another random name on my starknet Id that already has a domain then I lose the ownership of the domain that I overrode ?

File: src/naming/registration.cairo [hunk 1]
```diff
@@ -103,6 +104,33 @@ func assert_control_domain{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range
     return ();
 }
 
+// We might remove it in the future and keep a clientisde check
```
Comment (0xBenaparte) [inline src/naming/registration.cairo:107]: Erreur dans le commentaire

File: src/naming/registration.cairo [hunk 2]
```diff
@@ -103,6 +104,33 @@ func assert_control_domain{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range
...
+    let (sid_hashed_domain) = StarknetId.get_verifier_data(
+        contract_addr, starknet_id, 'name', naming_contract
+    );
+
+    with_attr error_message("This StarknetId already has a domain") {
+        // if a domain was written, check if it expired
+        if (sid_hashed_domain != 0) {
+            let (data) = _domain_data.read(sid_hashed_domain);
+            assert_le_felt(data.expiry, current_timestamp);
+            // because cairo is hell
```
Comment (0xBenaparte) [inline src/naming/registration.cairo:121]: Met moi une explication au moins mdrr

File: src/naming/registration.cairo [hunk 3]
```diff
@@ -103,6 +104,33 @@ func assert_control_domain{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range
...
+            tempvar syscall_ptr = syscall_ptr;
+            tempvar pedersen_ptr = pedersen_ptr;
+            tempvar range_check_ptr = range_check_ptr;
+        } else {
+            tempvar syscall_ptr = syscall_ptr;
```
Comment (0xBenaparte) [inline src/naming/registration.cairo:126]: Sachant que tu fais dans le if et le else tu peux pas le faire avant ou après la boucle ?

**Review by 0xBenaparte — CHANGES_REQUESTED** [review]
Juste explication sur le if else et un petit problème de commentaire !

**Comment by Th0rgal** [comment]
> So if I write another random name on my starknet Id that already has a domain then I lose the ownership of the domain that I overrode ?  

Yes that would be the idea if I removed the check. So we should check on the client side. To me it's like having a check in transfer() to see if you transfer your StarkNetID to the address 0 or a normal address. We should not do this contract side.

File: src/naming/registration.cairo (hunk 3 above)
Comment (Th0rgal) [reply src/naming/registration.cairo:126]: Nope là il s'agit d'unifier la position des pointeurs pour que les builtins soient corrects peu importe le branchement du programme. On aura plus ce genre de soucis en cairo 1.0. Tu peux essayer de faire des changements et compiler si tu veux check

**Comment by Th0rgal** [comment]
J'ai enlevé le check (et donc le commentaire avec). Dis moi si ça te va @0xBenaparte . Est-ce que tu peux confirmer qu'on check bien qu'une starnet id n'a pas de domaine sur le client ?

**Comment by Th0rgal** [comment]
replaced by #17

**Pull request closed without merging**